tests/test_generators.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+import unittest
+import tempfile
+import math
+import numpy as np
+from numpy import dtype
+
+from pupynere import netcdf_file, nc_generator
+
+VAR_NAME='temp'
+VAR_TYPE='>f4'
+VAR_VAL=math.pi
+
+n1dim = 4
+n2dim = 10
+n3dim = 8
+ranarr = 100.*np.zeros(shape=(n1dim,n2dim,n3dim))
+
+class TestGeneratorNonrecvars(unittest.TestCase):
+
+    def runTest(self):
+        f = netcdf_file(None, 'w')
+        temp = f.createVariable(VAR_NAME,VAR_TYPE)
+        temp.assignValue(VAR_VAL)
+
+        # Test filesize property of a virtual file with nonrecvars
+        assert f.filesize == 88
+
+        # Test generator
+        pipeline = nc_generator(f, _input())
+        with tempfile.NamedTemporaryFile(suffix=".nc") as fn:
+            for block in pipeline:
+                fn.write(block)
+
+            nc = netcdf_file(fn.name, 'r')
```

Comment:
Should this be outside of the `with tempfile...` block to ensure that the output gets flushed to the file, before you try to open it?